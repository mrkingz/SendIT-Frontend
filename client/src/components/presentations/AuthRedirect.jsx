import React from "react";
import { Link } from "react-router-dom";
import { PropTypes } from "prop-types";
import classnames from "classnames";

export const AuthRedirect = ({
  text,
  path,
  prompt,
  wrapperStyles,
  linkStyles
}) => {
  return (
    <div className="control-group">
      <div className={classnames(`auth-redirect ${wrapperStyles}`)}>
        <span>
          {prompt}
          <Link
            className={classnames(`bold auth-link ${linkStyles}`)}
            to={path}
          >
            {text}
          </Link>
        </span>
      </div>
    </div>
  );
};

AuthRedirect.defaultProps = {
  prompt: ""
};

AuthRedirect.propTypes = {
  text: PropTypes.string,
  path: PropTypes.string,
  prompt: PropTypes.string,
  styles: PropTypes.string,
  wrapperStyles: PropTypes.string,
  linkStyles: PropTypes.string
};
export default AuthRedirect;
