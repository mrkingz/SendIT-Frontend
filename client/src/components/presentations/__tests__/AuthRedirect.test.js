import React from "react";
import { shallow } from "enzyme";
import { AuthRedirect } from "../AuthRedirect";

const props = {
  text: "Message",
  linkStyles: "style",
  prompt: "prompt",
  wrapperStyles: "styles",
  path: "path"
};

const wrapper = shallow(<AuthRedirect {...props} />);

describe("Auth Redirect", () => {
  it("should render Auth Redirect", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
