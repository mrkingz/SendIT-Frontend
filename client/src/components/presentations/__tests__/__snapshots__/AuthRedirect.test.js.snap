// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Redirect should render Auth Redirect 1`] = `
<div
  className="control-group"
>
  <div
    className="auth-redirect styles"
  >
    <span>
      prompt
      <Link
        className="bold auth-link style"
        replace={false}
        to="path"
      >
        Message
      </Link>
    </span>
  </div>
</div>
`;
